presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/AbstractArrowMetadata.java

```diff
@@ -184,14 +186,21 @@ public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, Conn
             logger.debug("The value of the flight columnName is:- %s", columnName);
 
             Type type = getPrestoTypeFromArrowField(field);
-            columns.put(columnName, new ArrowColumnHandle(columnName, type));
+            columnHandles.put(columnName, new ArrowColumnHandle(columnName, type));
         }
-        return columns;
+        return columnHandles;
     }
 
     @Override
     public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns)
     {
+        if (!(table instanceof ArrowTableHandle)) {
+            throw new PrestoException(
+                    StandardErrorCode.INVALID_CAST_ARGUMENT,
+                    "Invalid table handle: Expected an instance of ArrowTableHandle but received "
+                            + table.getClass().getSimpleName() + ". Please check the table type being used.");
```

Comment:
User doesn't have an option to check table type. Error message doesn't need the string "Please check....."

Reply:
done